src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/set_interior_vehicle_data_request_test.cc

```diff
@@ -155,55 +157,58 @@ class SetInteriorVehicleDataRequestTest
 
 TEST_F(SetInteriorVehicleDataRequestTest,
        Execute_ValidWithoutReadOnlyParams_ExpectResendToHMI) {
+  using namespace rc_rpc_plugin::message_params;
+
   // Arrange
-  MessageSharedPtr mobile_message = CreateBasicMessage();
-  ns_smart_device_link::ns_smart_objects::SmartObject& msg_params =
+  auto mobile_message = CreateBasicMessage();
+  auto& msg_params =
       (*mobile_message)[application_manager::strings::msg_params];
-  msg_params[message_params::kModuleData][message_params::kModuleType] =
-      mobile_apis::ModuleType::CLIMATE;
-  smart_objects::SmartObject climate_control_data(smart_objects::SmartType_Map);
-  climate_control_data[message_params::kFanSpeed] = 10;
+  msg_params[kModuleData][kModuleType] = mobile_apis::ModuleType::CLIMATE;
+  auto& control_data = msg_params[kModuleData][kClimateControlData];
+  const uint64_t fan_speed = 10;
+  control_data[kFanSpeed] = fan_speed;
 
-  msg_params[message_params::kModuleData][message_params::kClimateControlData] =
-      climate_control_data;
   // Expectations
   EXPECT_CALL(mock_policy_handler_, CheckModule(kPolicyAppId, _))
-      .WillOnce(Return(rc_rpc_plugin::TypeAccess::kAllowed));
+      .WillOnce(Return(true));
 
   EXPECT_CALL(
       mock_rpc_service_,
       ManageHMICommand(
           HMIResultCodeIs(hmi_apis::FunctionID::RC_SetInteriorVehicleData), _))
       .WillOnce(Return(true));
   // Act
-  auto command =
-      CreateRCCommand<rc_rpc_plugin::commands::SetInteriorVehicleDataRequest>(
-          mobile_message);
+  auto command = CreateRCCommand<SetInteriorVehicleDataRequest>(mobile_message);
   ASSERT_TRUE(command->Init());
   command->Run();
+
+  EXPECT_TRUE(control_data.keyExists(kFanSpeed) &&
+              fan_speed == control_data[kFanSpeed].asUInt());
 }
 
 TEST_F(
     SetInteriorVehicleDataRequestTest,
     Execute_ValidWithSettableAndReadOnlyParams_ExpectCutReadOnlyAndResendToHMI) {
-  // Arrange
+  using namespace rc_rpc_plugin::message_params;
 
-  MessageSharedPtr mobile_message = CreateBasicMessage();
-  ns_smart_device_link::ns_smart_objects::SmartObject& msg_params =
+  // Arrange
+  auto mobile_message = CreateBasicMessage();
+  auto& msg_params =
       (*mobile_message)[application_manager::strings::msg_params];
-  msg_params[message_params::kModuleData][message_params::kModuleType] =
-      mobile_apis::ModuleType::RADIO;
-  smart_objects::SmartObject radio_control_data(smart_objects::SmartType_Map);
-  radio_control_data[message_params::kState] = true;
-  radio_control_data[message_params::kRadioEnable] = true;
-  msg_params[message_params::kModuleData][message_params::kRadioControlData] =
-      radio_control_data;
+  msg_params[kModuleData][kModuleType] = mobile_apis::ModuleType::RADIO;
+  auto& control_data = msg_params[kModuleData][kRadioControlData];
+  control_data[kState] = true;
+  const auto radio_enable = true;
```

Comment:
@mvorobio can't see any reason to use `auto` for explicit bool value.

Reply:
@AByzhynar please check 30f301a
There is no obvious benefit of `auto` instead of explicit `bool` in this case. But I think use of `auto` where it is possible at least helps think about objects not their types.